### DISCUSSION TITLE
Using MLflow for tracking experiments, packaging detectron2 model

### DISCUSSION THREAD

**Discussion by waelkht** [facebookresearch/detectron2#2629] [Q&A]
I was trying to use MLflow to track detectron2 experiments, log the performance metrics and package the model to deploy it later through MLflow API. However, the MLflow is not showing any metrics and the artifacts are not saved too.

Is there a way to use detectron2 with MLflow

This is how I usually log a PyTorch model, which is not working here
....
    mlflow.set_tracking_uri("http://localhost:5000")
    mlflow.set_experiment("Detecttron2_Object_detection")
.
.
.
with mlflow.start_run() as run:
      trainer.train()

**Comment by e-dzia** [comment]
I used a solution I found here and it worked: https://philenius.github.io/machine%20learning/2022/01/09/how-to-log-artifacts-metrics-and-parameters-of-your-detectron2-model-training-to-mlflow.html
